libraries/lib-ipc/internal/spinlock.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include <atomic>
+#include <thread>
+
+/**
+ * \brief Plain spinlock mutex implementation,
+ * can be used with std::lock_guard
+ */
+class spinlock
+{
+   std::atomic_flag flag = ATOMIC_FLAG_INIT;
+public:
+   ~spinlock() { }
+
+   void lock()
+   {
+      for(unsigned i = 0; flag.test_and_set(std::memory_order_acquire); ++i)
+         if(i & 1) std::this_thread::yield();
```

Comment:
This yields only every 2 failed tests.  I'm curious, why 2, not 1, or some other number.  Did some article recommend that?